src/api/app/views/webui/repositories/_repository_entry.html.haml

```diff
@@ -1,16 +1,53 @@
-      - if repository.architectures.empty?
-        %i No architecture selected
-      - else
-        - repository.architectures.pluck(:name).each do |architecture|
-          %span.badge.bg-primary= architecture
-    - if repository.is_dod_repository?
-      = render partial: 'dod_repository_card_content', locals: { project: project, repository: repository,
-                 available_architectures: available_architectures }
+= render(CardComponent.new) do |component|
+  - if repository.is_dod_repository?
+    = render partial: 'dod_repository_card_content', locals: { project: project, repository: repository,
+                available_architectures: available_architectures }
+    - component.with_delete_button do
+      - if user_can_modify
+        = link_to('#', title: 'Delete Repository', data: { 'bs-toggle': 'modal', 'bs-target': '#delete-repository',
+          action: destroy_repository_path(project: project, target: repository.name), repository: repository.name }) do
+          %i.fas.fa-times-circle.text-danger
+      - elsif User.session
+        .col
+          = link_to('#', title: 'Request Delete Repository', data: { 'bs-toggle': 'modal', 'bs-target': '#request-delete-repository',
+            action: project_remove_target_request_path(project: project, target: repository), repository: repository.name }) do
+            %i.fas.fa-user-times.text-danger
+    - if user_can_modify
+      - component.with_action do
+        = link_to('#', title: 'Add Download on Demand Source', data: { 'bs-toggle': 'modal',
+                                                                       'bs-target': "#add-dod-source-modal-#{repository.id}" }) do
+          %i.fas.fa-plus-circle.text-primary
+  - else
+    = render partial: 'repository_card_content', locals: { project: project, repository: repository,
+                download_url: download_url, user_can_modify: user_can_modify,
+                available_architectures: available_architectures }
+    - component.with_delete_button do
+      - if user_can_modify
+        = link_to('#', title: 'Delete Repository', data: { 'bs-toggle': 'modal', 'bs-target': '#delete-repository',
+          action: destroy_repository_path(project: project, target: repository.name), repository: repository.name }) do
+          %i.fas.fa-times-circle.text-danger
+      - elsif User.session
+        = link_to('#', title: 'Request Delete Repository', data: { 'bs-toggle': 'modal', 'bs-target': '#request-delete-repository',
+          action: project_remove_target_request_path(project: project, repository: repository.name), repository: repository.name }) do
+          %i.fas.fa-user-times.text-danger
+    - if user_can_modify
+      - component.with_action do
+        = link_to('#', title: 'Edit Repository',
+                      data: { 'bs-toggle': 'modal', 'bs-target': "#edit-repository-#{repository.id}" }) do
+          %i.fas.fa-edit.text-secondary
```

Comment:
![repository-card-after-with-text](https://user-images.githubusercontent.com/7080830/218122984-4c590d90-00b3-4904-874c-9db4e3785a10.png)

While I like it, it doesn't add too much information because we cannot add anything more than the first word of the action:
- Edit
- Add
- Download
which is self-explained by the icon. It would be much better if we could explain verbosely the whole thing
- Edit this repository
- Add a repository path
- Download repository
but we don't have all that space.

What do you say, should we keep the text anyway or just icons? I'm fine with both at the moment
@danidoni what do you think?

Reply:
What about adding it only for small screen layouts? I think we can benefit from the bigger click area on small devices that way... What do you think?